odoo/orm/models.py

```diff
@@ -247,23 +247,21 @@ def add_default(name, field):
                     setattr(self, name, field)
                     field.__set_name__(self, name)
 
-            add('id', Id(automatic=True))
-            add_default('display_name', Char(
-                string='Display Name', automatic=True,
-                compute='_compute_display_name',
-                search='_search_display_name',
-            ))
+            # make sure `id` is not overwritten
```

Comment:
```suggestion
            # make sure `id` type is not overwritten
```

Reply:
It is about the field not the type.

Reply:
Your error is only raised when the Id is not a instance of Field.Id. Then we can actually add your Id field to extend the BaseModel.id field right ? "overwritten" word is actually confusing IMO.

Reply:
Done